### PR TITLE
Cleanup error handling in solvers/qbf

### PR DESCRIPTION
**Pull request by xbauch**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 6a24c2c6b239e069a0e27005aa3de9b55d01fe57).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84858721

File: src/solvers/qbf/qbf_bdd_core.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ qbf_bdd_coret::~qbf_bdd_coret()
 
 tvt qbf_bdd_coret::l_get(literalt a) const
 {
-  assert(false);
+  UNREACHABLE;
   return tvt(false);
```
Comment (peterschrammel) [inline src/solvers/qbf/qbf_bdd_core.cpp]: Why is this required here and not below, eg in `m_get`?

File: src/solvers/qbf/qbf_bdd_core.cpp [hunk 2]
```diff
@@ -122,17 +122,18 @@ propt::resultt qbf_bdd_coret::prop_solve()
 
       *matrix=matrix->ExistAbstract(*bdd_variable_map[var]);
     }
-    else if(it->type==quantifiert::UNIVERSAL)
+    else
     {
+      INVARIANT(
+        it->type == quantifiert::UNIVERSAL,
+        "Only handles quantified variables.");
```
Comment (peterschrammel) [inline src/solvers/qbf/qbf_bdd_core.cpp]: start with lower case (several occurrences)

**Comment by xbauch** [comment]
Lower cases and full stops fixed, redundant return statements removed.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: fb13b743ff2963a836799d11508da0d6fae581d5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85159051

File: src/solvers/qbf/qbf_bdd_core.cpp [hunk 3]
```diff
@@ -310,7 +312,9 @@ const exprt qbf_bdd_certificatet::f_get(literalt l)
 
     // no cached function, so construct one
 
-    assert(model_bdds[l.var_no()]!=NULL);
+    INVARIANT(
+      model_bdds[l.var_no()] != NULL,
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_bdd_core.cpp]: While at it: please use `nullptr` instead of `NULL`

File: src/solvers/qbf/qbf_quantor.cpp [hunk 4]
```diff
@@ -8,9 +8,9 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "qbf_quantor.h"
 
-#include <cassert>
 #include <cstdlib>
 #include <fstream>
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_quantor.cpp]: Nit-pick: add a blank line before this include.

File: src/solvers/qbf/qbf_qube.cpp [hunk 5]
```diff
@@ -8,9 +8,9 @@ Author: CM Wintersteiger
 
 #include "qbf_qube.h"
 
-#include <cassert>
 #include <cstdlib>
 #include <fstream>
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_qube.cpp]: Nit-pick: add a blank line before this include.

File: src/solvers/qbf/qbf_qube_core.cpp [hunk 6]
```diff
@@ -8,10 +8,10 @@ Author: CM Wintersteiger
 
 #include "qbf_qube_core.h"
 
-#include <cassert>
 #include <cstdlib>
-#include <fstream>
 #include <cstring>
+#include <fstream>
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_qube_core.cpp]: Move one line down to group with the `mp_arith.h` include

File: src/solvers/qbf/qbf_qube_core.h [hunk 7]
```diff
@@ -11,6 +11,7 @@ Author: CM Wintersteiger
 #define CPROVER_SOLVERS_QBF_QBF_QUBE_CORE_H
 
 #include "qdimacs_core.h"
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_qube_core.h]: Nit-pick: add a blank line before this include.

File: src/solvers/qbf/qbf_skizzo.cpp [hunk 8]
```diff
@@ -8,9 +8,9 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "qbf_skizzo.h"
 
-#include <cassert>
 #include <cstdlib>
 #include <fstream>
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_skizzo.cpp]: Nit-pick: add a blank line before this include.

File: src/solvers/qbf/qbf_skizzo_core.cpp [hunk 9]
```diff
@@ -6,9 +6,8 @@ Author: CM Wintersteiger
 
 \*******************************************************************/
 
-
-#include <cassert>
 #include <fstream>
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_skizzo_core.cpp]: Nit-pick: add a blank line before this include.

File: src/solvers/qbf/qbf_skizzo_core.cpp [hunk 10]
```diff
@@ -234,8 +236,7 @@ bool qbf_skizzo_coret::get_certificate(void)
       e_lists=e_lists.substr(space+1);
     }
 
-    if(!result)
-      throw "existential mapping from sKizzo missing";
+    INVARIANT(!result, "existential mapping from sKizzo missing");
```
Comment (tautschnig) [inline src/solvers/qbf/qbf_skizzo_core.cpp]: Remove the negation, i.e., the invariant is `result`, not `!result`

File: src/solvers/qbf/qdimacs_cnf.cpp [hunk 11]
```diff
@@ -9,7 +9,7 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "qdimacs_cnf.h"
 
 #include <iostream>
-#include <cassert>
+#include <util/invariant.h>
```
Comment (tautschnig) [inline src/solvers/qbf/qdimacs_cnf.cpp]: Nit-pick: add a blank line before this include.

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Comment by xbauch** [comment]
Added blank lines before `util/invariant.h` includes, fixed the skizzo invariant condition typo.

**Review by tautschnig — APPROVED** [review]
Looks good to me, please squash and rebase.

**Comment by xbauch** [comment]
Squashed and rebased.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 706a529c9611142af1b71c4fb9408fc042e27304).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85264387

**Pull request merged**